pilot/cmd/pilot-agent/main.go

```diff
@@ -443,26 +428,14 @@ func init() {
 	proxyCmd.PersistentFlags().StringVar(&tokenManagerPlugin, "tokenManagerPlugin", tokenmanager.GoogleTokenExchange,
 		"Token provider specific plugin name.")
 	// Flags for proxy configuration
-	values := mesh.DefaultProxyConfig()
-	proxyCmd.PersistentFlags().StringVar(&configPath, "configPath", values.ConfigPath,
-		"Path to the generated configuration file directory")
-	proxyCmd.PersistentFlags().StringVar(&binaryPath, "binaryPath", values.BinaryPath,
-		"Path to the proxy binary")
-	proxyCmd.PersistentFlags().StringVar(&serviceCluster, "serviceCluster", values.ServiceCluster,
```

Comment:
isn't this different for every deployment? I am not sure how it can be configured in mesh config. I'd need to look closer at how it's used to be sure though

Reply:
IC, this is why the telemetry failed

Reply:
```
       - --serviceCluster
        {{ if ne "" (index .ObjectMeta.Labels "app") -}}
        - "{{ index .ObjectMeta.Labels `app` }}.$(POD_NAMESPACE)"
        {{ else -}}
        - "{{ valueOrDefault .DeploymentMeta.Name `istio-proxy` }}.{{ valueOrDefault .DeploymentMeta.Namespace `default` }}"
        {{ end -}}
```

For the above injection, pilot-agent can construct it with pod labels and namespace. 

But for vm proxy, we have to provide env's like `POD_NAMESPACE` and labels file, i donot think it's worthy.